src/kafunk/Utility/Buffer.fs

```diff
@@ -70,26 +73,28 @@ and BufferBound =
 
 module Buffer =
 
+  let getSize (buf: Buffer<'a>) = buf.Size
+
   let startBlockingWithConsumer 
     (capacity:int) 
     (batchSize:int) 
     (batchTimeMs:int)
     (timeIntervalMs:int)
     (consume:'a[] -> Async<unit>)
-    (blocking:int -> unit) : 'a -> bool =
+    (blocking:int -> unit) : Buffer<'a> * ('a -> bool) =
     let buf = new Buffer<'a> (BufferBound.BlockAfter capacity)
     buf.Consume (batchSize, batchTimeMs, timeIntervalMs, consume) |> Async.Start
     buf.Blocking |> Event.add blocking
-    buf.Add
+    buf, buf.Add
```

Comment:
TBH I think we can just drop this whole Buffer module and use the Buffer type directly from the ProducerBuffer module. This might make sense if we abstract the Buffer type at some point.